diff -Naur keystone-2013.2.b2.orig/bin/keystone-all keystone-2013.2.b2/bin/keystone-all
--- keystone-2013.2.b2.orig/bin/keystone-all	2013-07-18 19:01:20.000000000 +0000
+++ keystone-2013.2.b2/bin/keystone-all	2013-07-22 00:31:19.289679079 +0000
@@ -17,7 +17,6 @@
     sys.path.insert(0, possible_topdir)
 
 
-from paste import deploy
 import pbr.version
 
 from keystone.common import environment
@@ -26,6 +25,8 @@
 from keystone.openstack.common import gettextutils
 from keystone.openstack.common import importutils
 
+from paste import deploy
+
 
 CONF = config.CONF
 
diff -Naur keystone-2013.2.b2.orig/keystone/__init__.py keystone-2013.2.b2/keystone/__init__.py
--- keystone-2013.2.b2.orig/keystone/__init__.py	2013-07-22 00:29:30.734905980 +0000
+++ keystone-2013.2.b2/keystone/__init__.py	2013-07-22 00:32:08.954490625 +0000
@@ -0,0 +1,29 @@
+import sys
+import pkg_resources
+
+# If there is a conflicting non egg module,
+# i.e. an older standard system module installed,
+# then replace it with this requirement
+def replace_dist(requirement):
+    try:
+        return pkg_resources.require(requirement)
+    except pkg_resources.VersionConflict:
+        e = sys.exc_info()[1]
+        dist=e.args[0]
+        req=e.args[1]
+        if dist.key == req.key and not dist.location.endswith('.egg'):
+            del pkg_resources.working_set.by_key[dist.key]
+            # We assume there is no need to adjust sys.path
+            # and the associated pkg_resources.working_set.entries
+            return pkg_resources.require(requirement)
+
+replace_dist("WebOb >= 1.0")
+replace_dist("SQLAlchemy >= 0.6.3")
+replace_dist("Routes >= 1.12.3")
+
+replace_dist("PasteDeploy >= 1.5.0")
+# This hack is needed because replace_dist() results in
+# the standard paste module path being at the start of __path__.
+# TODO: See can we get pkg_resources to do the right thing directly
+import paste
+paste.__path__.insert(0, paste.__path__.pop(-1))
